nomad/structs/structs.go

```diff
@@ -2067,6 +2122,34 @@ type RequestedDevice struct {
 	Affinities []*Affinity
 }
 
+func (r *RequestedDevice) Equals(o *RequestedDevice) bool {
+	if r.Name == o.Name &&
+		r.Count == o.Count {
```

Comment:
You can avoid the large indention by returning false if names or counts don't match.